src/input/process/availability.rs

```diff
@@ -44,79 +74,125 @@ enum LimitType {
     time_slice_info: &TimeSliceInfo,
-) -> Result<HashMap<ProcessID, EnergyLimitsMap>>
+    milestone_years: &[u32],
+) -> Result<HashMap<ProcessID, ProcessEnergyLimitsMap>>
 where
     I: Iterator<Item = ProcessAvailabilityRaw>,
 {
     let mut map = HashMap::new();
-
     for record in iter {
-        let process_id = process_ids.get_id_by_str(&record.process_id)?;
-
-        ensure!(
-            record.value >= 0.0 && record.value <= 1.0,
-            "value for availability must be between 0 and 1 inclusive"
-        );
-
+        record.validate()?;
+
+        // Get process
+        let id = process_ids.get_id_by_str(&record.process_id)?;
+        let process = processes
+            .get(&id)
+            .with_context(|| format!("Process {id} not found"))?;
+
+        // Get regions
+        let process_regions = process.regions.clone();
+        let record_regions =
+            parse_region_str(&record.regions, &process_regions).with_context(|| {
+                format!("Invalid region for process {id}. Valid regions are {process_regions:?}")
+            })?;
+
+        // Get years
+        let process_year_range = &process.years;
+        let process_years: Vec<u32> = milestone_years
+            .iter()
+            .copied()
+            .filter(|year| process_year_range.contains(year))
+            .collect();
+        let record_years = parse_year_str(&record.year, &process_years).with_context(|| {
+            format!("Invalid year for process {id}. Valid years are {process_years:?}")
+        })?;
+
+        // Get timeslices
         let ts_selection = time_slice_info.get_selection(&record.time_slice)?;
 
-        let map = map.entry(process_id).or_insert_with(EnergyLimitsMap::new);
-
+        // Insert the energy limit into the map
+        let entry = map.entry(id).or_insert_with(ProcessEnergyLimitsMap::new);
         for (time_slice, ts_length) in time_slice_info.iter_selection(&ts_selection) {
-            // Calculate fraction of annual energy as availability multiplied by time slice length
-            // The resulting limits are max/min PAC energy produced/consumed in each timeslice per
-            // cap2act units of capacity
-            let value = record.value * ts_length;
-            let bounds = match record.limit_type {
-                LimitType::LowerBound => value..=f64::INFINITY,
-                LimitType::UpperBound => 0.0..=value,
-                LimitType::Equality => value..=value,
-            };
-
-            let existing = map.insert(time_slice.clone(), bounds.clone()).is_some();
-            ensure!(
-                !existing,
-                "Process availability entry covered by more than one time slice \
-                (process: {}, time slice: {})",
-                record.process_id,
-                time_slice
-            )
+            let bounds = record.to_bounds(ts_length);
+
+            for region in &record_regions {
+                for year in record_years.clone() {
```

Comment:
We don't need to clone the `Vec` every time:

```suggestion
                for year in record_years.iter().copied() {
```